x-pack/plugins/enterprise_search/server/routes/enterprise_search/engines.test.ts

```diff
@@ -26,22 +26,42 @@ describe('engines routes', () => {
   describe('GET /internal/enterprise_search/engines', () => {
     let mockRouter: MockRouter;
 
+    const mockClient = {
+      asCurrentUser: {
+        transport: {
+          request: jest.fn(),
+        },
+      },
+    };
     beforeEach(() => {
       jest.clearAllMocks();
+      const context = {
+        core: Promise.resolve({ elasticsearch: { client: mockClient } }),
+      } as unknown as jest.Mocked<RequestHandlerContext>;
+
       mockRouter = new MockRouter({
+        context,
         method: 'get',
         path: '/internal/enterprise_search/engines',
       });
-
       registerEnginesRoutes({
         ...mockDependencies,
         router: mockRouter.router,
       });
     });
 
-    it('creates a request to enterprise search', () => {
-      expect(mockRequestHandler.createRequest).toHaveBeenCalledWith({
-        path: '/api/engines',
+    it('List all engines creates a request to the ent-search-module', async () => {
```

Comment:
Nitpick: We're listing `search applications` now and no longer `engines` (I'm OK if this happens in a followup refactoring/renaming PR though)

Reply:
Can correct it in this PR as well. But I think there are other test classes that need this rename

Reply:
We can leave it for the later PR.